subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServices.java

```diff
@@ -79,7 +78,7 @@ BuildCacheServiceRegistration createDirectoryBuildCacheServiceRegistration() {
     @Override
     public void registerGradleServices(ServiceRegistration registration) {
         // Not build scoped because of dependency on GradleInternal for build path
-        registration.addProvider(new BuildCacheTaskServices());
+        registration.addProvider(new org.gradle.caching.internal.services.BuildCacheServices());
```

Comment:
Is there maybe a better name for one of the two  classes, so we don't have `BuildCacheServices` twice?